src/JustSaying.Models/Message.cs

```diff
@@ -7,11 +7,13 @@ public abstract class Message
         protected Message()
         {
             TimeStamp = DateTime.UtcNow;
-            Id = Guid.NewGuid();
+            TimeStampWithOffset = DateTimeOffset.Now;
+            Id = Guid.NewGuid();            
```

Comment:
I think it would be better to put this into a local, and then reference from there. Otherwise you have the possibility of having, albeit slight in magnitude, probably just a few ticks, differences in the value of the time in the two properties.

Something like:

```csharp
var now  = DateTimeOffset.Now;
TimeStampWithOffset = now;
TimeStamp = now.UtcDateTime;
```

Reply:
Should we perhaps call this something like UtcTimeStamp? Or TimeStampUtc?

That it has an offset is a .net implementation detail - the key bit is as you say, we are explicitly declaring it a UTC timestamp.

Reply:
Also, do we want to use DateTimeOffset.UtcNow? Or is this expected to be the servers Timex even if that's not in UTC?

Reply:
> Should we perhaps call this something like UtcTimeStamp? Or TimeStampUtc?
> 
> That it has an offset is a .net implementation detail - the key bit is as you say, we are explicitly declaring it a UTC timestamp.

Hmm I don't know - I think having an offset could be useful sometimes.  It gives you some extra informaion (what was the offset where this message was sent) without any loss.
We also can't guarantee it won't be overridden with a different offset.

Note that as currently written, it won't (necessarily) be "UTC" (i.e. offset 0)

Reply:
Renaming it from `TimeStamp` would be a breaking change.

I would agree that I'd prefer to just use UTC for both, rather than introduce two different ones that make assumptions about the time zone of the executing machine (which in prod for us will be UTC anyway).

I didn't bring that up initially though as I assumed there was offline conversation between Luke, Stu and Gareth that resulted in this PR using local times in the first place.

Reply:
Oh I see @gkinsman was referring to the existing timestamp.

The decision to use local was mine since, as I say it contains strictly more information.  Didn't discuss with Luke or Stu.

Reply:
Apologies if I wasn't clear - I was referring to the new field. If it's going to be a `DateTimeOffset.UtcNow`, then we may as well be clear in the naming that it's a UTC DateTimeOffset?

Reply:
If I had to choose I'd just have it as the UTC offset (so `+00:00`). It's still more informative and it's more-or-less explicit, it just lacks a non-zero offset value, which for for non-UI isn't really a problem IMHO.

As "+01:00" in the UK in the summer is just an inference of UK local time to us, it could equally be the time in the Canary Islands. Without specifying the time zone's _name_ you can't be 100% sure of what the offset is specifically referring to anyway.

Reply:
> If I had to choose I'd just have it as the UTC offset (so `+00:00`). It's still more informative and it's more-or-less explicit, it just lacks a non-zero offset value, which for for non-UI isn't really a problem IMHO.
> 
> As "+01:00" in the UK in the summer is just an inference of UK local time to us, it could equally be the time in the Canary Islands. Without specifying the time zone's _name_ you can't be 100% sure of what the offset is specifically referring to anyway.

Not sure I agree Martin.  The point is it could tells you what time it was for a relevant party at the time the performed the action, which is extra info.  I don't think the name of the timezone is important.

This is moot for us since as you say it will always be UTC and I'm happy to go either way, but local is my preference.  @stuart-lang any thoughts?

Reply:
I think we need to be a bit careful about adding fields to this base class, and only do so when we have a really good reason, and scenarios that make it really worthwhile.

To add some more data, let's consider Australia's timezones. Our servers are located in `ap-southeast-2`, which is in Sydney. Australia has three timezones, some of which apply DST, and others that do not, at different times of the year. Given that Menulog is available all over Australia, the local timezone that an order is placed in could be quite different to the timezone that the server that handled that request is in. In this case, using this new field for anything would be quite risky, as the local time could be either ahead or behind the server time - we would need to account for it in any usages of the field.

In the above scenario, if we did actually want to capture the local time where the order was placed, I think it would make more sense to be explicit and to use a `LocalOrderPlacedAt` DateTimeOffset field on whichever messages that fulfill that flow, which would be populated from an incoming request. 

I'm not flat out suggesting we don't do this, just that I think we need some solid scenarios and use cases to support its addition.

Reply:
The key reason for this field isn't to capture something about timezone, it's to ensure we have an unambiguous timestamp available on the message.  For JE it is really moot whether we use `Now` or `UtcNow` because the servers are set to UTC anyway.

Reply:
The offset is only useful for caring about that specific message at that point in time. You can't use the offset to infer future things from it, as the offset may change during the life of the order, which again makes me wonder how useful it is to have a non-UTC value.

The only truly unambiguous way to represent the time zone is to use its name (e.g. `Europe/London` in IANA format).

Sure you can say for a specific event that it happened at a wall-clock time of 09:52 to the user, but you can't rely on the offset to compute other timestamps relative to it (e.g. what is the wall-clock time an hour later), which is what makes me think you might as well just encode it as `+00:00` to make it clear it's UTC.

Reply:
It potentially makes things easier like "find all the events that happened between 9 and 5 local time".  I have a very weak opinion on this though.  If the consensus is for UTC I'll do that.  And it's 2 on 1 so far. 

Be good to get your thougts on my [comment below](https://github.com/justeat/JustSaying/pull/849#discussion_r562702600) as well.